resilience4j-spring/src/main/java/io/github/resilience4j/ratelimiter/configure/RateLimiterAspect.java

```diff
@@ -38,74 +43,108 @@
+	public Object rateLimiterAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, RateLimiter limitedService) throws Throwable {
+		RateLimiter targetService = limitedService;
+		Method method = ((MethodSignature) proceedingJoinPoint.getSignature()).getMethod();
+		String methodName = method.getDeclaringClass().getName() + "#" + method.getName();
+		if (targetService == null) {
+			targetService = getRateLimiterAnnotation(proceedingJoinPoint);
+		}
+		String name = targetService.name();
+		Class<?> returnType = method.getReturnType();
+		io.github.resilience4j.ratelimiter.RateLimiter rateLimiter = getOrCreateRateLimiter(methodName, name);
+		if (rateLimiterAspectExtList != null && !rateLimiterAspectExtList.isEmpty()) {
+			for (RateLimiterAspectExt rateLimiterAspectExt : rateLimiterAspectExtList) {
+				if (rateLimiterAspectExt.canHandleReturnType(returnType)) {
+					return rateLimiterAspectExt.handle(proceedingJoinPoint, rateLimiter, methodName);
+				}
+			}
+		} else if (CompletionStage.class.isAssignableFrom(returnType)) {
+			return handleJoinPointCompletableFuture(proceedingJoinPoint, rateLimiter, methodName);
+		}
+		return handleJoinPoint(proceedingJoinPoint, rateLimiter, methodName);
+	}
+
+	private io.github.resilience4j.ratelimiter.RateLimiter getOrCreateRateLimiter(String methodName, String name) {
+		io.github.resilience4j.ratelimiter.RateLimiter rateLimiter = rateLimiterRegistry.rateLimiter(name);
+
+		if (logger.isDebugEnabled()) {
+			RateLimiterConfig rateLimiterConfig = rateLimiter.getRateLimiterConfig();
+			logger.debug(
+					RATE_LIMITER_RECEIVED,
+					name, rateLimiterConfig.getLimitRefreshPeriod(), rateLimiterConfig.getLimitForPeriod(),
+					rateLimiterConfig.getTimeoutDuration(), methodName
+			);
+		}
+
+		return rateLimiter;
+	}
+
+	private RateLimiter getRateLimiterAnnotation(ProceedingJoinPoint proceedingJoinPoint) {
+		return AnnotationExtractor.extract(proceedingJoinPoint.getTarget().getClass(), RateLimiter.class);
+	}
+
+	private Object handleJoinPoint(ProceedingJoinPoint proceedingJoinPoint,
+	                               io.github.resilience4j.ratelimiter.RateLimiter rateLimiter, String methodName)
+			throws Throwable {
+		try {
+			io.github.resilience4j.ratelimiter.RateLimiter.waitForPermission(rateLimiter);
+			return proceedingJoinPoint.proceed();
+		} catch (Exception exception) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("Invocation of method '" + methodName + "' failed!", exception);
+			}
+			throw exception;
+		}
+	}
+
+	/**
+	 * handle the asynchronous completable future flow
+	 *
+	 * @param proceedingJoinPoint AOPJoinPoint
+	 * @param rateLimiter         configured rate limiter
+	 * @param methodName          bulkhead method name
+	 * @return CompletionStage
+	 * @throws Throwable
+	 */
+	private Object handleJoinPointCompletableFuture(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.ratelimiter.RateLimiter rateLimiter, String methodName) {
```

Comment:
i guess handleAsync would give a feeling it can handle other types of Async like Futures..ect, the method name here is more specific to what it support which CompletableFuture only .